src/column/reader.rs

```diff
@@ -881,7 +1036,7 @@ mod tests {
 
       // Generate the current page
 
-      let mut pb = DataPageBuilderImpl::new(desc.clone(), levels_per_page as u32, use_v2);
+      let mut pb = DataPageBuilderImpl::new(desc.clone(), num_values_cur_page as u32, use_v2);
```

Comment:
We should be recording actual values written, not just `levels_per_page`, because for `test_read_batch` it becomes important (okay for others).

Reply:
Nice! can you also update the doc for `DataPageBuilderImpl::new` to note that the `num_values` is actually the number of non-null values?

Also nit: line width exceeds 90 characters.